lib/app/pull.go

```diff
@@ -232,13 +252,15 @@ func (r Puller) pullApp(loc loc.Locator) error {
 	}
 	logger := r.WithField("app", loc)
 	if app != nil && !upsert {
-		if r.SkipIfExists {
+		err = r.onConflict(loc)
+		if utils.IsAbortError(err) {
 			return nil
 		}
-		logger.Info("Application already exists.")
-		return trace.AlreadyExists("application %v already exists", loc)
+		if err != nil {
+			return trace.Wrap(err)
+		}
 	}
-	logger.Info("Pull application.")
+	logger.Debug("Pull application.")
```

Comment:
Nit: IMO it was quite helpful to see this log entry without having to enable debug mode. Same for "pull package" above. Unless this is a duplicate and it's already logged elsewhere?